### PR TITLE
Fix proxy settings types when reading the configurations

### PR DESCRIPTION
**Pull request by EhabY**

### CODE REVIEW

File: test/unit/api/utils.test.ts [hunk 1]
```diff
@@ -225,5 +225,23 @@ describe("createHttpAgent", () => {
 				await agent.getProxyForUrl("https://fallback.example.com", mockRequest),
 			).toBe(proxy);
 		});
+
+		it("trims and joins multiple http.noProxy entries", async () => {
+			const cfg = new MockConfigurationProvider();
+			cfg.set("http.proxy", proxy);
+			cfg.set("http.noProxy", [" first.example.com ", "second.example.com "]);
```
Comment (mtojek) [inline test/unit/api/utils.test.ts:232]: nit: maybe another test case with empty `[]` ?

File: src/api/utils.ts [hunk 2]
```diff
@@ -28,8 +28,8 @@ export async function createHttpAgent(
 ): Promise<ProxyAgent> {
 	const insecure = cfg.get<boolean>("coder.insecure", false);
 	const proxyStrictSSL = cfg.get<boolean>("http.proxyStrictSSL", true);
-	const proxyAuthorization = cfg.get<string>("http.proxyAuthorization");
-	const httpNoProxy = cfg.get<string>("http.noProxy");
+	const proxyAuthorization = cfg.get<string | null>("http.proxyAuthorization");
```
Comment (mtojek) [inline src/api/utils.ts:31]: hmm.. I don't see any test to exercise this change. Do we need one?

Comment (EhabY) [reply src/api/utils.ts:31]: TypeScript already enforces the type when it knows it so I think it's fine

File: test/unit/api/utils.test.ts (hunk 1 above)
Comment (EhabY) [reply test/unit/api/utils.test.ts:232]: True, done!
